packages/amplify-graphql-api-construct-tests/src/__tests__/backends/sql-configurable-stack/app.ts

```diff
@@ -0,0 +1,310 @@
+        },
+      ];
+    };
+  };
+  dbConnectionConfig: SqlModelDataSourceDbConnectionConfig;
+}
+
+interface StackConfig {
+  /**
+   * The AppSync GraphQL schema, provided as string for AmplifyGraphqlApi Construct definition.
+   */
+  schema: string;
+
+  /**
+   * The AuthorizationMode type for AmplifyGraphqlApi Construct.
+   */
+  authMode?: AUTH_MODE;
+
+  /**
+   * If true, disable Cognito User Pool/Auth resources creation and only use API Key auth in sandbox mode.
+   */
+  useSandbox?: boolean;
+
+  /**
+   * Cognito User Pool groups to create when provisioning the User Pool.
+   *
+   * **NOTE**
+   * Provide at least two group names for setup and testing purposes.
+   */
+  userGroups?: string[];
+
+  /**
+   * The OIDC options/config when using OIDC AuthorizationMode for AmplifyGraphqlApi Construct.
+   *
+   * @property {Record<string, string>} [triggers] - UserPoolTriggers for Cognito User Pool when provisioning the User Pool as OIDC provider.
+   * - key: trigger name e.g. 'preTokenGeneration'
+   * - value: the lambda function code inlined as a string
+   *
+   * **NOTE**
+   * - Only applicable when AuthorizationMode is set to OIDC.
+   * - Currently only supports Cognito User Pools as the simulated OIDC provider for E2E test.
+   * - Currently only supports JavaScript as the lambda function code, with Node.js runtime version 18.x.
+   * - Inline code needs to export the handler function as `handler` as `index.handler` would be used as the handler path.
+   */
+  oidcOptions?: {
+    triggers: Record<string, string>;
+  };
+}
+
+const createApiDefinition = (): IAmplifyGraphqlDefinition => {
+  const schema = stackConfig.schema ?? SCHEMA.DEFAULT;
+
+  return AmplifyGraphqlDefinition.fromString(schema, {
+    name: dbDetails.dbConfig.strategyName,
+    dbType: dbDetails.dbConfig.dbType,
+    vpcConfiguration: {
+      vpcId: dbDetails.dbConfig.vpcConfig.vpcId,
+      securityGroupIds: dbDetails.dbConfig.vpcConfig.securityGroupIds,
+      subnetAvailabilityZoneConfig: dbDetails.dbConfig.vpcConfig.subnetAvailabilityZones,
+    },
+    dbConnectionConfig: {
+      ...dbDetails.dbConnectionConfig,
+    },
+    sqlLambdaProvisionedConcurrencyConfig: {
+      provisionedConcurrentExecutions: 2,
+    },
+  });
+};
+
+const createAuthorizationModes = (): AuthorizationModes => {
+  const auth = stackConfig.authMode ?? AUTH_MODE.API_KEY;
+  let authorizationModes: AuthorizationModes;
+
+  switch (auth) {
+    case AUTH_MODE.API_KEY: {
+      authorizationModes = {
+        defaultAuthorizationMode: 'API_KEY',
+        apiKeyConfig: { expires: Duration.days(7) },
+      };
+
+      break;
+    }
+    case AUTH_MODE.AMAZON_COGNITO_USER_POOLS: {
+      const { userPool, userPoolClient } = createUserPool(dbDetails.dbConfig.dbName);
+
+      authorizationModes = {
+        defaultAuthorizationMode: 'AMAZON_COGNITO_USER_POOLS',
+        userPoolConfig: {
+          userPool,
+        },
+        apiKeyConfig: { expires: Duration.days(2) },
```

Comment:
Do we need the API key to be valid for 2 days? Especially since we teardown these apps post test run, I would recommend setting it to 1 hr to begin with.

Reply:
Line 75 of file `authorization-modes.ts` in `amplify-graphql-api-construct` enforces the `Duration` would be converted to days with following definition: `apiKeyExpirationDays: authMode.expires.toDays()`

Setting expiration to hours would cause CDK deployment to fail with error message:
- "Error: '1 hours' cannot be converted into a whole number of days."

We could adjust the construct time conversion if needed to give test a more tight time boundary.

Reply:
> We could adjust the construct time conversion if needed to give test a more tight time boundary.

That might be the right way to go forward here.